src/backend/distributed/commands/function.c

```diff
@@ -1303,18 +1287,92 @@ PostprocessCreateFunctionStmt(Node *node, const char *queryString)
 		return NIL;
 	}
 
-	ObjectAddress address = GetObjectAddressFromParseTree((Node *) stmt, false);
-	EnsureDependenciesExistOnAllNodes(&address);
+	ObjectAddress functionAddress = GetObjectAddressFromParseTree((Node *) stmt, false);
 
-	List *commands = list_make4(DISABLE_DDL_PROPAGATION,
-								GetFunctionDDLCommand(address.objectId, true),
-								GetFunctionAlterOwnerCommand(address.objectId),
-								ENABLE_DDL_PROPAGATION);
+	if (IsObjectAddressOwnedByExtension(&functionAddress, NULL))
+	{
+		return NIL;
+	}
+
+	ObjectAddress *undistributableDependency = GetUndistributableRelationDependency(
+		&functionAddress);
+	if (undistributableDependency != NULL)
+	{
+		if (SupportedDependencyByCitus(undistributableDependency))
```

Comment:
maybe add a comment saying that this check should have been (a) valid for all objects not only for functions (b) should check for all types of objects no only for relations. 

We do this limited check for now as functions are more likely to be used with (a) and (b), and we want to scope it for now

Reply:
Added